fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraWebACUserSecurityContext.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.auth.common;
+
+import java.security.Principal;
+
+import org.apache.commons.lang.StringUtils;
+import org.modeshape.jcr.security.SecurityContext;
+import org.modeshape.jcr.value.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The security context for Fedora WebAC servlet users. These users are not
+ * necessarily authenticated by the container, i.e. users may include the
+ * general public. This security context delegates all access decisions to the
+ * configured authorization delegate.
+ *
+ * @author mohideen
+ */
+public class FedoraWebACUserSecurityContext extends FedoraUserSecurityContext {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(FedoraWebACUserSecurityContext.class);
+
+    /**
+     * Constructs a new security context.
+     *
+     * @param userPrincipal the user principal associated with this security
+     *        context
+     * @param fad the authorization delegate
+     */
+    protected FedoraWebACUserSecurityContext(final Principal userPrincipal,
+                                        final FedoraAuthorizationDelegate fad) {
+        super(userPrincipal, fad);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see SecurityContext#hasRole(String)
+     */
+    @Override
+    public final boolean hasRole(final String roleName) {
+        // Under this custom PEP regime, all users have modeshape read and write
+        // roles.
+        if ("http://www.w3.org/ns/auth/acl#Read".equals(roleName)) {
+            return true;
+        } else if ("http://www.w3.org/ns/auth/acl#Write".equals(roleName)) {
+            return true;
+        } else if ("http://www.w3.org/ns/auth/acl#Append".equals(roleName)) {
+            return true;
+        } else if ("http://www.w3.org/ns/auth/acl#Control".equals(roleName)) {
```

Comment:
As mentioned above, we do not want to introduce "web-ac" concepts into the core codebase. We should figure out how to instead move this class into the fcrepo-module-auth-webac project.